.github/workflows/javascript_integration.yml

```diff
@@ -0,0 +1,48 @@
+name: Javascript Tests
+
+on: 
+  push:
+    branches:
+      - main
+      
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix: 
+        platform: [windows-latest, macos-latest, ubuntu-latest]
+        node-version: ["v20", "v22", "v24"]
+
+
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+    - name: Check out repository code
+      uses: actions/checkout@v4
+
+    - name: Use Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+
+    - name: Restore npm cache
+      uses: actions/cache@v4
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json')}}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Install dependencies
+      run: npm ci
+
+      
+    - name: Build platform
+      run: npm run build
+      
+    - name: Test platform
+      run: npm run sync-version
```

Comment:
```suggestion
```

This won't have an effect, and anyways this will be something we run by hand when we do a release. Please remove, and then we can merge this!